exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/grpc/DefaultGrpcExporterBuilder.java

```diff
@@ -110,6 +111,10 @@ public GrpcExporterBuilder<T> setClientTls(byte[] privateKeyPem, byte[] certific
 
   @Override
   public DefaultGrpcExporterBuilder<T> addHeader(String key, String value) {
+    if (key.equals("host")) {
+      authorityOverride = value;
+      return this;
```

Comment:
I don't know enough about this, but will people be surprised if it doesn't end up as an actual header?

Reply:
[Host](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host) is a header in HTTP - not sure if it plays any role in gRPC.

Reply:
It does actually end up in a header, `: authority`, just grpc doesn't support setting it through the metadata API but has this separate API.

Reply:
Ah but it wouldn't be named `host`. AFAIK it is not expected to set host on a http/2 request. For example, if proxies get both with different values, it will ignore the host value when converting to HTTP/1, moving the value from authority into host, from what I've seen.

In practice I couldn't find any way to set both with okhttp anyways, it will always accept host and either set it as host on http/1 or authority on http/2 with no way of setting host on the latter, probably for the above reason. So we'll probably want to just keep that behavior consistent across all the exporters.

Reply:
Is there any problem with using `if (key.equals(":authority")) { ... }` instead?

Reply:
OkHttp doesn't accept `: authority` and silently ignores it